### DISCUSSION TITLE
STEP200 (Arduino Zero Ethernet Board)

### DISCUSSION THREAD

**Discussion by 7sorkr** [ponoor/step-series-support#26] [Q&A]
I happened to find the board below on the web

![OAC866ET-1](https://user-images.githubusercontent.com/32606306/187112924-35d88cba-1b1b-408d-aa6c-8b5e550c5508.jpg)
https://www.makerfabs.com/wiki/index.php?title=Maduino_Zero_Ethernet

Does the ponoor step200 document seem to meet everything??!! (W5500 + SAMD21 + SPI)

I want to connect X-Nucleo-IHM02A1 with the board linked below.

Arduino-->Ethernet-->X-Nucleo-IHM02A1 (existing connection)

Can I know which pin I must connect to when connecting to X-Nucleo-IHM02A1 on Ethernet?

**Comment by kanta** [comment]
@7sorkr Interesting board! This should work with IHM02A1.

For the connection, you can check [this list](https://ponoor.com/en/docs/step100-200/assemble/step200/).
I think connecting `D8`, `D11`, `D12`, `D13`, `A2`, `3.3V`, `GND` should be okay.

I found another but similar board, but it's out of stock.
https://www.seeedstudio.com/Squame-Ethernet-Arduino-W5500-Ethernet-Board-p-4823.html

**Reply by 7sorkr** [reply]
> @7sorkr재미있는 판! 이것은 IHM02A1에서 작동해야 합니다.
> 
> [연결의 경우 이 목록](https://ponoor.com/en/docs/step100-200/assemble/step200/) 을 확인할 수 있습니다 . , , , , , , 연결 은 괜찮을 것 같아요 .`D8``D11``D12``D13``A2``3.3V``GND`
> 
> 비슷한 보드를 또 찾았는데 품절입니다. https://www.seeedstudio.com/Squame-Ethernet-Arduino-W5500-Ethernet-Board-p-4823.html

Thank you always.

I've been watching the STEP series and Arduino

It's fun

**Reply by 7sorkr** [reply]
> @7sorkr재미있는 판! 이것은 IHM02A1에서 작동해야 합니다.
> 
> [연결의 경우 이 목록](https://ponoor.com/en/docs/step100-200/assemble/step200/) 을 확인할 수 있습니다 . , , , , , , 연결 은 괜찮을 것 같아요 .`D8``D11``D12``D13``A2``3.3V``GND`
> 
> 비슷한 보드를 또 찾았는데 품절입니다. https://www.seeedstudio.com/Squame-Ethernet-Arduino-W5500-Ethernet-Board-p-4823.html

Yes, I connected all of them, but it's not working

Do I have to change something when I sketch Arduino?

Currently, it's the same as Arduino Zero

**Reply by 7sorkr** [reply]
> @7sorkr재미있는 판! 이것은 IHM02A1에서 작동해야 합니다.
> 
> [연결의 경우 이 목록](https://ponoor.com/en/docs/step100-200/assemble/step200/) 을 확인할 수 있습니다 . , , , , , , 연결 은 괜찮을 것 같아요 .`D8``D11``D12``D13``A2``3.3V``GND`
> 
> 비슷한 보드를 또 찾았는데 품절입니다. https://www.seeedstudio.com/Squame-Ethernet-Arduino-W5500-Ethernet-Board-p-4823.html

It says MKR1000 (Com Port)

**Reply by kanta** [reply]
If your device is recognized as MKR1000, then you may need some modifications,since the pin assignment is different from Zero.
I think the easiest way is to burn the Zero boot loader from SWD port...

**Reply by 7sorkr** [reply]
swd bootloader Can you give me a link or method to refer to..?

I've tried many things so far, but it doesn't work.

**Reply by kanta** [reply]
Here is the [document ](https://paper.dropbox.com/doc/STEP400-programming-the-bootloader--BoSQn5M3xqt9kG7UjZ5zwE8xAQ-SLwrCIULBbT5QjQP0hd2p) about it. I used this document for explaining this process to the PCB manufacture of STEP400/STEP800.
I have only tested with [ATMEL-ICE-BASIC](https://www.microchip.com/en-us/development-tool/ATATMEL-ICE).

**Reply by 7sorkr** [reply]
> 여기에 관한 [문서](https://paper.dropbox.com/doc/STEP400-programming-the-bootloader--BoSQn5M3xqt9kG7UjZ5zwE8xAQ-SLwrCIULBbT5QjQP0hd2p) 가 있습니다. 이 문서를 사용하여 STEP400/STEP800의 PCB 제조에 ​​이 과정을 설명했습니다. [ATMEL-ICE-BASIC으로](https://www.microchip.com/en-us/development-tool/ATATMEL-ICE)
> 만 테스트했습니다 .

I bought atmel-ice.

The Arduino IDE is a zero boot loader.

Compilation and upload is complete.

However, it repeats usb recognition and disconnection

And this message comes out of the serial monitor.

![3F16DCAF-72D9-4705-84DD-2429F7F6A8AB](https://user-images.githubusercontent.com/32606306/188672365-733d3d68-ce03-45ca-a534-aba3083c5fbc.jpeg)

![F2CC95EA-F7C6-422C-9A1C-093643951648](https://user-images.githubusercontent.com/32606306/188673455-0c927245-326a-4717-88b6-2ba15abc4da1.jpeg)

**Reply by 7sorkr** [reply]
https://www.makerfabs.com/wiki/images/a/a0/MaduinoZeroEthernetV01_SCH.pdf

**Reply by 7sorkr** [reply]
[maudino_zero_etherne.zip](https://github.com/ponoor/step-series-support/files/9502800/maudino_zero_etherne.zip)

Microchip studio > Ctrl+Shift+P > Atmel-ICE (ATSAMD21G18A) > SWD (Apply click) > Fuses > Memories > hex file import > Program

Atmel ICE

**Reply by kanta** [reply]
Messages from the serial monitor seem to be good. 
I guess this part of the schematic could be important.
![image](https://user-images.githubusercontent.com/1097902/188883302-a8a6b92e-cbe6-461c-b138-01626879b357.png)

These connections are different from the original Arduino Zero. The problem is you can't change `W5500_CS` in `boardsDef.h` so far. I will modify the code and let you know when I finish it.

**Reply by kanta** [reply]
@7sorkr What are you using for uploading the STEP200 firmware? With VisualStudio Code, or Arduino IDE?

**Reply by 7sorkr** [reply]
Arduino IDE

**Reply by 7sorkr** [reply]
> Messages from the serial monitor seem to be good. I guess this part of the schematic could be important. ![image](https://user-images.githubusercontent.com/1097902/188883302-a8a6b92e-cbe6-461c-b138-01626879b357.png)
> 
> These connections are different from the original Arduino Zero. The problem is you can't change `W5500_CS` in `boardsDef.h` so far. I will modify the code and let you know when I finish it.

I haven't tried it in days!

Thanks a lot.

**Reply by kanta** [reply]
@7sorkr I have updated some codes, could you try it? The new codes are on the [develop branch](https://github.com/ponoor/step-series-universal-firmware/tree/develop) of the repo, or you can download the zipped file. 
[step-series-universal-firmware.zip](https://github.com/ponoor/step-series-support/files/9522482/step-series-universal-firmware.zip)

You need some modifications in `boardsDef.h` to fit your board.

1. comment out Line18 `#define STEP400_R1` and uncomment Line23 `// #define STEP200_R1`.
2. uncomment L302 `/ #define W5500_RESET_PIN_OPERATION_STATE    HIGH`.
3. uncomment L321 `// #define PIN_W5500_RESET A3` and change `A3` to `9u`.
4. change L322 as `#define PIN_W5500_CS    3u`.
5. change L304 as `#define ledPin 5u`. Actually `5u` can be changed any other unused pins.

I hope this works and let me know the result!

**Reply by 7sorkr** [reply]
Yes, I see.

Is PIN OUT the same..?

D8, D11, D12, D13, A2, 3.3V, GND

**Reply by kanta** [reply]
Yes, should be same.

**Reply by 7sorkr** [reply]
The SD memory used by the STEP200 was inserted.

When I removed the SD memory from the board, I got the /booted command.

I changed the information below from the globals.cpp file.

IP Address
myIp(192, 168, 0, 100),
myIp_from_config,
destIp(192, 168, 0, 10),
dns(168, 126, 63, 1)
Gateway (192, 168, 0, 1)

-----------------------------------
If you give the /setDestIp command, the statement below will continue to repeat.
![loop_message](https://user-images.githubusercontent.com/32606306/189042126-c76efa64-e92c-4531-b5ea-827f01469bab.PNG)

------------------------------------
The information below is captured by Serial Monitor.
![status](https://user-images.githubusercontent.com/32606306/189042159-0851569f-5606-4689-95ff-4df5f612da1f.PNG)

![Config](https://user-images.githubusercontent.com/32606306/189042190-6272a754-2739-4ff2-bf4f-63eb25998df4.PNG)

**Reply by kanta** [reply]
Thanks for the report. Did you insert the SD card when you got status/configurations from the Serial Monitor?

**Reply by 7sorkr** [reply]
SD memory has been removed.

-------------------------------------------

Once SD memory is inserted, the USB will stick and fall again and again.

As soon as you remove SD memory, the /booted command appears.

**Reply by kanta** [reply]
It seems there are 2 problems.

1. microSD card connection
2. communication with motor drivers

Let's start with the problem 2. Are you providing a power supply to your nucleo board (IHM02A1)? If not, connect the power supply and reset the Arduino. Then could you post what the Serial Monitor said when you send `s`?

**Reply by 7sorkr** [reply]
![status_02](https://user-images.githubusercontent.com/32606306/189058350-bdbd4ee9-0ff2-44b5-ae88-94412d0533a6.PNG)

**Reply by 7sorkr** [reply]
![1F5E34E5-3D77-48F2-A23B-8C01ECF8D4C9](https://user-images.githubusercontent.com/32606306/189061210-3b446296-2ebd-4f3d-ac0e-edb4415d4066.jpeg)

**Reply by kanta** [reply]
Thanks, I have to go now and I will check tomorrow. There are many possible reasons so it could be difficult to solve the problem without physical board...

**Reply by 7sorkr** [reply]
Arduino SAMD Board
If you compile and upload to version 1.6.21, the over current does not come out.

/booted (ok)
/setDestIp (ok)
/getversion (ok)

Motor 1 and 2 are connected. (Check with the s command)

If X-Nucleo-IHM02A1 is turned off in the current state,
The "current over" command is coming out like before
However, if you reconnect the power, the current over command is lost.

The get command works well.
The set command is not allowed.

In conclusion, it's almost the same as before

------------------------------------------------------------------------
============== Current Status ==============
-------------- Firmware --------------
Firmware name : STEP200_R1_UNIVERSAL
Firmware version : 1.2.3
Compile date : Sep 13 2022, 00:18:38
Applicable config version : 1.2
Loaded config version : -1.0 [CONFIG_VERSION_NOTLOADED]
Config and product match : No
-------------- DIP Switch --------------
BIN : 0000 0001
DEC : 1
-------------- Ethernet --------------
Ethernet hardware status: 3 -EthernetW5500
Ethernet link status: 1 -LinkON
isDestIpSet : No
Destination Ip : 192.168.0.66
Current My Ip : 192.168.0.22
Current MAC address : 60:95:CE:10:06:01
Current outPort : 50101
-------------- microSD --------------
SD library initialize succeeded : No
SD config file open succeeded : No
SD config JSON parse succeeded : No
-------------- Motor Driver --------------
Motor driver SPI connection established : Yes
Driver ID#1
	ABS_POS: 0xFFE10000 (-2031616step)
	EL_POS: 0x00 (Fullstep:0, Microstep:0)
	MARK: 0xFFE30000 (-1900544step)
	SPEED: 0x40000 (3906.4step/s)
	ACC: 0x500 (9313.3step/s/s)
	DEC: 0x600 (11175.9step/s/s)
	MAX_SPEED: 0x300 (11718.8step/s)
	MIN_SPEED: 0x800 (487.4step/s)
	FS_SPD: 0x100 (3913.9step/s)
	KVAL_HOLD: 0x29 (41)
	KVAL_RUN: 0x2A (42)
	KVAL_ACC: 0x2B (43)
	KVAL_DEC: 0x2C (44)
	INT_SPD: 0x2D00 (11520)
	ST_SLP: 0x2E (46)
	FN_SLP_ACC: 0x2F (47)
	FN_SLP_DEC: 0x30 (48)
	K_THERM: 0x31 (2.5)
	ADC_OUT: 0x32 (50)
	OCD_TH: 0x03 (1500.0mA)
	STALL_TH: 0x34 (1656.2mA)
	STEP_MODE: 0x06 (6)
	ALARM_EN: 0x37 (0011 0111)
	CONFIG: 0x3800
	  OSC_SEL: 0
	  EXT_CLK: 0
	  SW_MODE: 0
	  EN_VSCOMP: 0
	  OC_SD: 0
	  POW_SR: 2
	  F_PWM_DEC: 0
	  F_PWM_INT: 1
	STATUS: 0xD000
	  High impedance state : No
	  BUSY : Yes
	  Motor direction : Reverse
	  Motor status : Stopped
	  UVLO (Undervoltage lock out) : Yes
	  Thermal status : Device shutsown
	  OCD (Overcurent detection) : No
	  Stalled : Yes
	  SW_F: 0 -HOME senser input open.
Driver ID#2
	ABS_POS: 0x00 (0step)
	EL_POS: 0x00 (Fullstep:0, Microstep:0)
	MARK: 0x00 (0step)
	SPEED: 0x00 (3906.4step/s)
	ACC: 0x00 (9313.3step/s/s)
	DEC: 0x00 (11175.9step/s/s)
	MAX_SPEED: 0x00 (11718.8step/s)
	MIN_SPEED: 0x00 (487.4step/s)
	FS_SPD: 0x00 (3913.9step/s)
	KVAL_HOLD: 0x00 (0)
	KVAL_RUN: 0x00 (0)
	KVAL_ACC: 0x00 (0)
	KVAL_DEC: 0x00 (0)
	INT_SPD: 0x00 (0)
	ST_SLP: 0x00 (0)
	FN_SLP_ACC: 0x00 (0)
	KVAL_RUN: 0x00 (0)
	KVAL_ACC: 0x00 (0)
	KVAL_DEC: 0x00 (0)
	INT_SPD: 0x00 (0)
	ST_SLP: 0x00 (0)
	FN_SLP_ACC: 0x00 (0)
	FN_SLP_DEC: 0x00 (0)
	K_THERM: 0x00 (1.0)
	ADC_OUT: 0x00 (0)
	OCD_TH: 0x00 (375.0mA)
	STALL_TH: 0x00 (31.2mA)
	STEP_MODE: 0x00 (0)
	ALARM_EN: 0x00 (0000 0000)
	CONFIG: 0x00
	  OSC_SEL: 0
	  EXT_CLK: 0
	  SW_MODE: 0
	  EN_VSCOMP: 0
	  OC_SD: 0
	  POW_SR: 0
	  F_PWM_DEC: 0
	  F_PWM_INT: 0
	STATUS: 0xD000
	  High impedance state : No
	  BUSY : Yes
	  Motor direction : Reverse
	  Motor status : Stopped
	  UVLO (Undervoltage lock out) : Yes
	  Thermal status : Device shutsown
	  OCD (Overcurent detection) : No
	  Stalled : Yes
	  SW_F: 0 -HOME senser input open.
-------------- Modes --------------
Servo Mode :  No, No
Homing status : 0, 0

**Reply by kanta** [reply]
@7sorkr Did you check with the original Arduino Zero + Ethernet Shield + Nucleo board? If so, the firmware is working well on my end, with Arduino IDE 1.8.19. How about try with the newest version of Arduino IDE? 

And I recommend to supply the motor power before the Arduino boot, otherwise driver setting commands won't be sent properly.

Also I want to know which are you working with, Arduino Zero or Arduino M0?

**Reply by 7sorkr** [reply]
Nucleo + bottom image
Arduino IDE Version 2.0.0.-rc9.4

I can't make it
If Kanta buys the board and succeeds, it will be a success

![image](https://user-images.githubusercontent.com/32606306/189902488-c593d412-6327-4027-afa3-dcd35e9f42ea.png)

**Reply by kanta** [reply]
I see, I thought you switched to the original Arduino Zero board but now I understand the problem is happened on this red board.

I compared between the schematics of the original [Arduino Zero](https://cdn-shop.adafruit.com/product-files/2843/Arduino-Zero-schematic.pdf) and the [Maduino Zero Ethernet](https://www.makerfabs.com/wiki/images/a/a0/MaduinoZeroEthernetV01_SCH.pdf), and they have same pin assignments and they should be fully compatible.

From the `Current Status` outputs, there are some strange things in it.
1. Values from the first motor seems random.
2. Second motor's values are all zero.

I guess there is a mistake around SPI connection for the motor drivers. Could you check the connection again for `D8`, `A2`, `D11`, `D12`, and `D13`?

Purchasing this board will be expensive when combined with shipping costs, and it will be difficult to find time for debugging right now. Since the support of STEP100/STEP200 is a completely volunteer based, I am sorry to say that testing with Maduino will be on my list of things to do in the future.

**Reply by kanta** [reply]
@7sorkr I found a big mistake in the code which I sent. This could make Ethernet connection unstable.
I'm not sure this is the reason of your problem but here is the fixed version.
https://github.com/ponoor/step-series-universal-firmware/releases/tag/v1.2.4

**Reply by 7sorkr** [reply]
If you go with the original Arduino Zero,
The IHM02A11 LED lights up and the 3 LED lights up every time you give a signal.

If you proceed with the Marudino zero ethernet integrated board,
IHM02A1. The lights keep on coming on in numbers 1 and 2
There's no LED response to the signal
I blink every time I signal the RJ45 terminal

The symptoms are the same as before.
Should I say no to this board?

It's frustrating because it seems like it's going to work but it's ^^

![1FE993D0-91AB-4859-A2C3-DA572407ABF8](https://user-images.githubusercontent.com/32606306/192091655-626f116f-f19b-4e7b-a1fb-55b8b4be720b.jpeg)

**Reply by kanta** [reply]
@7sorkr I understand your feeling. Programming hardware is always talking really long time to fix a small problem. My resources are limited but I will try to help to solve your problem.

Great to know your X-Nucleo board is working with original Arduino Zero. So X-Nucleo should be okay.
And you have already succeeded to communicate via OSC, so Ethernet part is also ok.

Then, your problem might be around SPI connection between Marduino and X-Nucleo board. To dive into details, could you share your STEP200 board definition in `boardsDef.h`? It should be around Line291 to Line326.

**Reply by 7sorkr** [reply]
#elif defined(STEP200_R1)
    #define PRODUCT_NAME    "STEP200"
    #define FIRMWARE_NAME   "STEP200_R1_UNIVERSAL"
    const uint8_t applicableConfigVersion[2] = {1,2};
    #define NUM_OF_MOTOR (2)
    #define HAVE_SD
    // #define HAVE_LIMIT_GPIO
    // #define HAVE_DIP_SW
    #define DRIVER_L6470
    #define DRIVER_OSC INT_16MHZ
    #define DRIVER_RESET_PIN_OPERATION_STATE   HIGH
    #define W5500_RESET_PIN_OPERATION_STATE    HIGH // ENABLE: Operation
    // pins
    #define ledPin 5u
#ifdef MZERO_WITH_ZERO_BOOTLOADER
    #define PIN_SD_CS   2u
#else
    #define PIN_SD_CS	4u
#endif
    // #define PIN_SD_DETECT   A4
    #define PIN_DRIVER_MISO 12u	// 
    #define PIN_DRIVER_MOSI 11u	// 
    #define PIN_DRIVER_SCK	13u	// 
    #define EPIO_DRIVER_MISO    PIO_SERCOM
    #define EPIO_DRIVER_MOSI    PIO_SERCOM
    #define EPIO_DRIVER_SCK     PIO_SERCOM
    #define PAD_DRIVER_SPI  SPI_PAD_0_SCK_1
    #define PAD_DRIVER_RX   SERCOM_RX_PAD_3
    #define PIN_DRIVER_CS   A2
    #define PIN_DRIVER_RESET    8u
    #define PIN_W5500_RESET 9u
    #define PIN_W5500_CS    3u
    #define DRIVER_SERCOM sercom1

    #define ID  1
#endif

Thank you from the bottom of my heart

**Reply by kanta** [reply]
@7sorkr Thanks for the code, everything seems to be ok. 
As I mentioned in the [document](https://ponoor.com/en/docs/step100-200/assemble/step200/), `PIN_DRIVER_RESET ` (or `STBY/RST`) is assigned to `D4` of the X-Nucleo board. Could you make sure the connection? This pin should be connected with `D8` of Arduino Zero or Maaduino.

**Reply by 7sorkr** [reply]
I connected only the pin connection you answered to NUCLEO.

![D3CA9BBE-E382-40A7-B389-C192A7AC504C](https://user-images.githubusercontent.com/32606306/192290909-b5f86e0b-fb99-47fc-9fef-6a975fcd4685.jpeg)

![83B4BDEA-AF01-423E-A1C1-1ACB34C48274](https://user-images.githubusercontent.com/32606306/192291486-8898b63a-99ae-4525-a6c3-d6af8f6b2e96.jpeg)

**Reply by kanta** [reply]
@7sorkr Thanks for the image. Could you show me the Marduino side just in case?
I tested the connection with jump wires with Arduino M0 (Arduino Zero bootloader), and it is working well.
![image](https://user-images.githubusercontent.com/1097902/192407660-6da63f84-99c2-48a5-8217-354c35c4956a.png)

I have double checked the schematics of Marduino and Arduino M0, but there are no differences between their pin assignments. So it should work...

**Reply by 7sorkr** [reply]
![C7B749BB-6249-42C0-9A3C-61B3FC409B1A](https://user-images.githubusercontent.com/32606306/192430014-1b7e0cfc-9cd2-4145-a5ef-196cb6a07932.jpeg)
![DA69CD5E-EA0F-474A-B16B-6C6BE34992C9](https://user-images.githubusercontent.com/32606306/192430032-0384cb37-08ef-45fd-a3fc-d3e2336329b2.jpeg)

**Reply by 7sorkr** [reply]
![75955AD8-628F-46D1-A0C4-475845E7CDF4](https://user-images.githubusercontent.com/32606306/192430471-d51e4b5f-5da0-4229-b7ea-122ae73c1a08.jpeg)

**Reply by kanta** [reply]
@7sorkr Thanks, the Marduino connection seems to be ok too.
Then... what we can do is to observe the each signals with an oscilloscope. This will help to find the reason, a software problem or a hardware problem. Do you have any kind of that?

**Reply by 7sorkr** [reply]
There is no oscilloscope.
You can check the open circuit with the tester. (HIOKI 3280-10F)

If there is, what should I do?

**Reply by kanta** [reply]
@7sorkr If you have an oscilloscope, you can observe SPI signal at A2, D11, D12, D13. These signals are quite fast and can't check with multimeters.
But you can check the connection between your Marduino and X-Nucleo board with your tester.

By the way I decided to buy one Marduino for debug. It will take maximum 1 month for delivery so pls wait for that.

**Reply by 7sorkr** [reply]
I hope a month goes by quickly

**Reply by kanta** [reply]
Hi @7sorkr , My Maarduino arrived, and I tried it out.

I found `A2` pin is attached some components in Marduino internally. Probably this design intends to detect a power supply from USB but the 0.1uF capacitor will destroy the CS signal of SPI for the X-Nucleo.
![image](https://user-images.githubusercontent.com/1097902/195285439-b072b212-bbf5-4f3b-998f-e3970e652396.png)

The proper CS signal:
<img src="https://user-images.githubusercontent.com/1097902/195286652-664b0098-1be7-4725-abba-4cdb0d330422.png" width="50%" />

The destroyed CS signal at `A2` pin of Marduino:
<img src="https://user-images.githubusercontent.com/1097902/195287009-31161b14-8601-4b60-9a88-9be54c3dedf6.png" width="50%" />


So change `A2` pin to an another one, like `A3`.
```C++
#define PIN_DRIVER_CS A3
```

You also need to change A2 to A3 for a jumpwire for your Marduino.
Then I made it to work!
![image](https://user-images.githubusercontent.com/1097902/195287795-adc51f16-986b-45d3-8cf6-0f642dac76d9.png)

**Reply by kanta** [reply]
@7sorkr For your information, you can find a tested code from `Maaruino-test` branch of the repo.
https://github.com/ponoor/step-series-universal-firmware/blob/Mardunio-test/src/boardsDef.h

**Comment by 7sorkr** [comment]


**Reply by kanta** [reply]
Unfortunately the STEP200 code won't work for this board because it has different MCU.

**Comment by 7sorkr** [comment]
STEP200 Complete

1U Rackcase

2 x Stepper Motor 
2 x HOME Sensor
2 x LIMIT Sensor

Handle the deep switch with a jumper cable

Once again, thank you Kanta.

![12CCBC5B-9292-4257-957F-01274195B848](https://user-images.githubusercontent.com/32606306/188130454-53d1ab46-a419-4b3e-bbc3-075fefb833c4.jpeg)

**Reply by kanta** [reply]
You're welcome @7sorkr ! I'm very happy to see STEP200 in action!!
By the way, would 1U be too short to fit the STEP200?

**Reply by 7sorkr** [reply]
It's not easy

Is there a good way?

![493D8998-0D6D-400E-859C-F10B49328CB4](https://user-images.githubusercontent.com/32606306/188172081-44a40332-3a0d-4324-875f-c549c519395c.jpeg)

**Reply by kanta** [reply]
Oh I see, this is how you made it. I think there is no easy way to reduce it's height... Probably using a 2U case is the fastest solution.

**Reply by 7sorkr** [reply]
https://www.st.com/resource/en/user_manual/dm00237629-getting-started-with-xnucleoihm02a1-twoaxis-stepper-motor-driver-expansion-board-based-on-l6470-for-stm32-nucleo-stmicroelectronics.pdf
